scripts/maintenance/import_projects_as_template.py

```diff
@@ -0,0 +1,153 @@
+        f"importing project {project_file}",
+    )
+    path = "/projects%3Aimport"
+    files = {"fileName": open(project_file, mode="rb")}
+    r = await client.post(path, files=files, timeout=20)
+    r.raise_for_status()
+    typer.secho(
+        f"project {project_file} imported, received uuid is {r.json()['data']}",
+        fg=typer.colors.YELLOW,
+    )
+    return r.json()["data"]["uuid"]
+
+
+async def rename_project(client: AsyncClient, project_uuid: str, project_name: str):
+    path = f"/projects/{project_uuid}"
+    r = await client.get(path)
+    r.raise_for_status()
+    typer.secho(
+        f"project {project_uuid} received, current name is {r.json()['data']['name']}",
+    )
+
+    modified_project = r.json()["data"]
+    modified_project["name"] = project_name
+
+    typer.secho(
+        f"renaming project {project_uuid} to {project_name}",
+        fg=typer.colors.YELLOW,
+    )
+
+    r = await client.put(path, json=modified_project)
+    r.raise_for_status()
+    typer.secho(
+        f"project {project_uuid} renamed",
+        fg=typer.colors.YELLOW,
+    )
+
+
+async def publish_as_template(client: AsyncClient, project_uuid: str) -> str:
+    path = f"/projects/{project_uuid}"
+    r = await client.get(path)
+    r.raise_for_status()
+    typer.secho(
+        f"project {project_uuid} received, current name is {r.json()['data']['name']}",
+    )
+
+    # set access rights
+    modified_project = r.json()["data"]
+    modified_project["accessRights"].update(
+        {"1": {"read": True, "write": False, "delete": False}}
+    )
+
+    path = "/projects"
+    typer.secho(
+        f"publishing project {project_uuid} as template...",
+        fg=typer.colors.YELLOW,
+    )
+    r = await client.post(
+        path, params={"as_template": project_uuid}, json=modified_project
+    )
+    r.raise_for_status()
+    typer.secho(
+        f"project published as template {r.json()['data']['uuid']}",
+        fg=typer.colors.YELLOW,
+    )
+    return r.json()["data"]["uuid"]
+
+
+async def import_project_as_template(
+    endpoint: URL,
+    username: EmailStr,
+    password: SecretStr,
+    project_file: Path,
+    project_name: str,
+    share_with_gid: int,
+) -> int:
+    try:
+        async with AsyncClient(base_url=endpoint.join("v0")) as client:
+            await login_user(client, username, password)
+            project_uuid = await import_project(client, project_file)
+            await rename_project(client, project_uuid, project_name)
+            template_uuid = await publish_as_template(client, project_uuid)
+            typer.secho(
+                f"project published as template! uuid [{template_uuid}]",
+                fg=typer.colors.BRIGHT_WHITE,
+            )
+
+    except Exception as exc:  # pylint: disable=broad-except
+        typer.secho(f"Unexpected issue: {exc}", fg=typer.colors.RED, err=True)
+        return 1
+
+    return 0
+
+
+def main(
+    endpoint: str,
+    username: str,
```

Comment:
I wonder if typer would validate and convert parameters if they are annotated``endpoint: URL, username: EmailStr, password:SecretStr``. Otherwise, one could use the [pydantic validator](https://pydantic-docs.helpmanual.io/usage/validation_decorator/) ?

Reply:
I tried.. and you get an error a la "this is not yet supported"